draft-ietf-quic-http.md

```diff
@@ -97,38 +97,82 @@ defined in {{!RFC2119}}.
 
 # QUIC advertisement
 
-A server advertises that it can speak HTTP/QUIC via the Alt-Svc HTTP response 
-header. It does so by including the header in any response sent over a non-QUIC 
-(e.g. HTTP/2) connection: 
+A server advertises that it can speak HTTP/QUIC via the Alt-Svc ({{!RFC7838}}) 
+HTTP response header, using the ALPN token defined in 
+{{connection-establishment}}. 
 
-   Alt-Svc: quic=":443"
+Thus, a server could indicate in an HTTP/1.1 or HTTP/2 response that HTTP/QUIC
+was available on UDP port 443 by including the following header in any
+response:
 
-In addition, the list of QUIC versions supported by the server can be specified
-by the v= parameter.  For example, if a server supported both version 33 and 34
-it would specify the following header:
+    Alt-Svc: hq=":443"
 
-   Alt-Svc: quic=":443"; v="34,33"
+## QUIC version hints {#alt-svc-version-hint}
 
-On receipt of this header, a client may attempt to establish a QUIC connection
+This document defines the "v" parameter for Alt-Svc, which is used to provide
+version-negotiation hints to HTTP/QUIC clients. Syntax:
+
+    v = quoted-versions
+    quoted-versions = DQUOTE *( version-string "," ) version-string DQUOTE
+    version-string = token; percent-encoded QUIC version
+
+For example, if a server supported both version Q033 and Q034 it would specify
+the following header:
+
+    Alt-Svc: hq=":443"; v="Q034,Q033"
+
+Where multiple versions are listed, the order of the values reflects the
+server's preference (with the first value being the most preferred version).
+
+QUIC versions are four-octet sequences with no additional constraints on format. 
+Octets not allowed in tokens ({{!RFC7230}}, Section 3.2.6) MUST be 
+percent-encoded as per Section 2.1 of {{!RFC3986}}. Consequently, the octet 
+representing the percent character ("%", hex 25) MUST be percent-encoded as 
+well. 
+
+On receipt of this header, a client MAY attempt to establish a QUIC connection
 on port 443 and, if successful, send HTTP requests using the mapping described
-in this document.
+in this document.  Servers SHOULD list only versions which they support, but MAY
```

Comment:
This seems to refer to the example specifically.  but we're now a few paragraphs down.  Either speak generically "the client MAY attempt to establish a QUIC connection on the specified port" or refer to the example directly.